frontendUiRoot/src/main/java/com/virtuslab/gitmachete/frontend/ui/root/GitMachetePanel.java

```diff
@@ -61,8 +61,13 @@ public GitMachetePanel(Project project) {
   private ActionToolbar createGitMacheteVerticalToolbar() {
     DefaultActionGroup gitMacheteActions = new DefaultActionGroup();
 
-    RefreshGitMacheteStatusAction refreshGitMacheteStatusAction = new RefreshGitMacheteStatusAction();
-    ActionManager.getInstance().registerAction(ActionIDs.ACTION_REFRESH, refreshGitMacheteStatusAction);
+    // This check is needed as action register is shared between multiple running IDE instances
```

Comment:
Hmm actually... does the same concern apply to any other of our actions beyond refresh? Or more generally, for what actions do we need this check and for what actions we don't?

Reply:
So there are two ways to register actions. By code (like here) or by `plugin.xml`. 

Reference: 
https://www.jetbrains.org/intellij/sdk/docs/basics/action_system.html

The refresh action is (and was) implemented here. The slide-out action requires to call it after it is done. Current implementation and (probably too strict?) relation between the panel and refresh disallow registering the refresh action via plugin.xml. That is why I used in-code registering. 

Actually, the reregistered action issue that happened here and for the sake of consistency we could consider moving all actions to `plugin.xml`.

Reply:
https://github.com/VirtusLab/git-machete-intellij-plugin/issues/220